### ISSUE TITLE
 java.lang.NoClassDefFoundError: Failed resolution of: Lokhttp3/internal/Util;

### ISSUE THREAD

**Issue by razaghimahdi** [GetStream/stream-video-android#734]
**SDK version**
- 0.2.2 && 0.2.1

On these version, I try to make a call, by this code:
```kotlin
    val client = StreamVideoBuilder(
            context = applicationContext,
            apiKey = "hd8szvscpxvd", // demo API key
            geo = GEO.GlobalEdgeNetwork,
            user = user,
            token = userToken,
        ).build()

        val call = client.call("default", callId)

       lifecycleScope.launch {
            val result = call.join(create = true)
            result.onError {
                Toast.makeText(applicationContext, "uh oh $it", Toast.LENGTH_SHORT).show()
            }
        }

```

my app would crash and it says:

java.lang.NoClassDefFoundError: Failed resolution of: Lokhttp3/internal/Util;
                                                                                                    	at io.getstream.video.android.core.CameraManager.setup$stream_video_android_core_release(MediaManager.kt:504)
                                                                                                    	at io.getstream.video.android.core.CameraManager.enable(MediaManager.kt:360)
                                                                                                    	at io.getstream.video.android.core.CameraManager.setEnabled(MediaManager.kt:387)
                                                                                                    	at io.getstream.video.android.core.CameraManager.setEnabled$default(MediaManager.kt:385)
                                                                                                    	at io.getstream.video.android.core.call.RtcSession.connectRtc(RtcSession.kt:521)
                                                                                                    	at io.getstream.video.android.core.call.RtcSession.connect(RtcSession.kt:372)
                                                                                                    	at io.getstream.video.android.core.call.RtcSession$connect$1.invokeSuspend(Unknown Source:14)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loop(Looper.java:236)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8057)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
                                                                                                    	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@2706457, Dispatchers.Main.immediate]
                                                                                                    Caused by: java.lang.ClassNotFoundException: Didn't find class "okhttp3.internal.Util" on path: DexPathList[[dex file "/data/data/com.montreux.mylife/code_cache/.overlay/base.apk/classes20.dex", zip file "/data/app/~~5PUw0rU2qXw3ROZHwl4aOg==/com.montreux.mylife-IWhgr9qVSxFm7AzdIPsiTA==/base.apk"],nativeLibraryDirectories=[/data/app/~~5PUw0rU2qXw3ROZHwl4aOg==/com.montreux.mylife-IWhgr9qVSxFm7AzdIPsiTA==/lib/arm64, /data/app/~~5PUw0rU2qXw3ROZHwl4aOg==/com.montreux.mylife-IWhgr9qVSxFm7AzdIPsiTA==/base.apk!/lib/arm64-v8a, /system/lib64, /system/system_ext/lib64]]
                                                                                                    	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:207)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
                                                                                                    	... 16 more

**Comment by tschellenbach** [comment]
thank you for reporting this. note that during the beta we're taking calls with customers to help them integrate and setup shared slack channels. did you already talk to our team & setup a shared slack?

our team will resolve this in the next release

**Comment by skydoves** [comment]
Hey @razaghimahdi, thanks for reporting this issue. We will fix this issue very soon in the next release (0.2.3).
In the meantime, would you check out if the snapshot version (`0.2.2-20230805.155650-4`) works well? You can see how to include our snapshot version following our guide: https://getstream.io/video/docs/android/basics/installation/#snapshot-builds

**Comment by razaghimahdi** [comment]
> thank you for reporting this. note that during the beta we're taking calls with customers to help them integrate and setup shared slack channels. did you already talk to our team & setup a shared slack?
> 
> our team will resolve this in the next release

thanks for your response, I didn't talk to your team, but next time I will.

**Comment by razaghimahdi** [comment]
> Hey @razaghimahdi, thanks for reporting this issue. We will fix this issue very soon in the next release (0.2.3). In the meantime, would you check out if the snapshot version (`0.2.2-20230805.155650-4`) works well? You can see how to include our snapshot version following our guide: https://getstream.io/video/docs/android/basics/installation/#snapshot-builds

hey man, thanks for your response, well it seems this version works, thanks for letting me know, may I ask how long it would takes to release new version?

**Comment by skydoves** [comment]
@razaghimahdi Thanks for checking the snapshot version. We're planning to release a stable version this week. I will keep you informed about the new updates.

**Comment by DanielNovak** [comment]
@razaghimahdi Are you maybe using OkHttp 5.0 (alpha) in your test project? Do you have it listed in your build.gradle? Our SDK is using OkHttp 4.x and it's not binary compatible. In this case the OkHttp Util class has moved to a different package - into [UtilCommon](https://github.com/square/okhttp/blob/parent-5.0.0-alpha.10/okhttp/src/commonMain/kotlin/okhttp3/internal/-UtilCommon.kt). But we will look how to guard against this.

**Comment by razaghimahdi** [comment]
> OkHttp

good point man, yeah actually it's 5.0.0-alpha.7, so what exact version u suggest? again tnx for the response

**Comment by DanielNovak** [comment]
We recommend to use the latest stable OkHttp release which is 4.11.0 (https://github.com/square/okhttp#releases). OkHttp 5.x is only in alpha now and has breaking changes that make it binary uncompatible and generally I would recommend to not use it for production until it's stable. You will probably also run into compatibility problems with other libraries with 5.x.

4.11.0 works with all our SDK releases. If you need 5.x then it will work with the 0.2.2-x SDK snapshot releases or you can wait for next stable 0.2.3.

**Comment by skydoves** [comment]
@razaghimahdi We've released version 0.3.0. If you don't face any issues with OkHttp 5.0.0-x with our 0.3.0, you can still use them together, but we'd recommend using OkHttp 4.11.0 still yet. Thank you for reporting this issue. Now we will close this ticket and will keep you updated if there are any changes. Thanks a lot!

**Comment by razaghimahdi** [comment]
> @razaghimahdi We've released version 0.3.0. If you don't face any issues with OkHttp 5.0.0-x with our 0.3.0, you can still use them together, but we'd recommend using OkHttp 4.11.0 still yet. Thank you for reporting this issue. Now we will close this ticket and will keep you updated if there are any changes. Thanks a lot!

wow, well that is a good news, thanks for the update. I'll definitely take your recommendation, again thanks❤
